helpers/plugins.js

```diff
@@ -0,0 +1,26 @@
+const pluginSets = new Map();
+
+export function getPlugin(setKey, pluginKey) {
+	const pluginSet = pluginSets.get(setKey);
```

Comment:
Should this `throw` if the set doesn't exist instead of returning `undefined`?

If this does end up being a `tryGet`, I'd suggest returning `null` if the plugin can't be found.

Reply:
Hmmm, the cases I've been considering so far would always have a respective plugin unless misconfigured. 

But let's say we have some file-type plugins for something, and we're given a file that we don't have a plugin for. In that case, a consumer might try to get the plugin keyed by extension or mime-type, and fallback if necessary. Alternatively, the consumer could get all the plugins, and do the `find` themselves, or iterate over them calling `tryDoSomething()`.

With that in mind, I sort of lean towards `tryGetPluginByKey`, returning `null` if not found. But I could be convinced to go the error route, because as mentioned, the consumer has other means to the same end - it's really just sugar.

Reply:
Definitely sugar yeah. As is this whole method and key thing anyway. Yeah I'm good with whatever!

Reply:
Resolving since we chose to rename method.